torch/_subclasses/fake_impls.py

```diff
@@ -269,6 +268,11 @@ def unique2(
         # Without symints/symfloats, cannot handle this
         raise DynamicOutputShapeException(func)
 
+    if dim is None:
+        arg_dim = arg
+    else:
+        arg_dim = arg.new_empty((arg.shape[dim],))
```

Comment:
While I can believe you get correct results this way, this is written in a needlessly circuitous way. When writing a meta, you should prefer to directly allocate the result tensors, do not allocate intermediate tensors that do not actually get returned

Reply:
Addressed in e040b23